agent/containers/images/visualizers/run.py

```diff
@@ -0,0 +1,128 @@
+"""
+Starts by starting up grafana server, as well as any collectors (prometheus, pmproxy) if needed.
+
+Then sets up all grafana plugins, data sources, and dashboards through grafana API.
+Chooses what to upload/enable based off VIS_TYPE environment variable.
+
+If VIS_TYPE is 'live' (default for live-metric-visualizer):
+    - Node exporter/DCGM dashboards will be uploaded and enabled
+    - Prometheus data source will be configured
+    - Grafana-pcp plugin will be enabled
+    - PCP redis and vector datasources will be configured
+    - PCP default dashboards will be enabled
+If VIS_TYPE is 'prom' (default for prom-graf-visualizer):
+    - Node exporter/DCGM dashboards will be uploaded and enabled
+    - Prometheus data source will be configured
+If VIS_TYPE is 'pcp' (default for soon-to-come pcp-graf-visualizer):
+    - Grafana-pcp plugin will be enabled
+    - PCP redis and vector datasources will be configured
+    - PCP default dashboards will be enabled
+"""
+
+import os
+import json
+import requests
+import time
+import subprocess
+
+# Start the collector (if needed)
+collector_cmd = os.environ["COLLECTOR"]
+if not collector_cmd == ":":
```

Comment:
The `":"` seems like a bit of a "magic number" that's not really necessary here (you could have omitted the environment variable or set it to `""`), and a bit of documentation at least in the README might be in order here, but I won't worry about it too much. Maybe consider cleaning this up when you add the PCP visualizer path?

Reply:
Filed https://github.com/distributed-system-analysis/pbench/issues/2078 to track.